internal/pkg/gateway/api.go

```diff
@@ -432,43 +432,42 @@ func (gs *Server) Submit(ctx context.Context, request *gp.SubmitRequest) (*gp.Su
 	for _, index := range rand.Perm(len(orderers)) {
 		orderer := orderers[index]
 		logger.Infow("Sending transaction to orderer", "txID", request.TransactionId, "endpoint", orderer.address)
-		err := gs.broadcast(ctx, orderer, txn)
-		if err == nil {
+		response, err := gs.broadcast(ctx, orderer, txn)
+		if err != nil {
+			errDetails = append(errDetails, errorDetail(orderer.endpointConfig, err.Error()))
+		}
+		status := response.GetStatus()
+		if status == common.Status_SUCCESS {
 			return &gp.SubmitResponse{}, nil
 		}
```

Comment:
`err` will be non-nil only on some gRPC connection/invocation error, and in this case `response` will be nil. GetStatus() for a nil `response` will return the zero value for the status so the code below works OK but the behaviour may not be immediately obvious. I wonder if, on a non-nil `err`, it would be clearer to have an explicit log and continue (or use an else-branch). The following code would then only run for a non-nil `response` and the behaviour (and log messages) may be more obvious. What do you think?

```suggestion
		response, err := gs.broadcast(ctx, orderer, txn)
		if err != nil {
			errDetails = append(errDetails, errorDetail(orderer.endpointConfig, err.Error()))
			logger.Warnw("Error sending transaction to orderer", "txID", request.TransactionId, "endpoint", orderer.address, "err", err)
			continue
		}

		status := response.GetStatus()
		if status == common.Status_SUCCESS {
			return &gp.SubmitResponse{}, nil
		}
		
		logger.Warnw("Unsuccessful response sending transaction to orderer", "txID", request.TransactionId, "endpoint", orderer.address, "status", status, "info", response.GetInfo())
```